tools/ctl/ctl.go

```diff
@@ -216,6 +217,47 @@ var getResults = &cobra.Command{
 	},
 }
 
+type PackageWorker interface {
+	Run(ctx context.Context, bar *pb.ProgressBar, out chan schema.Verdict, in chan benchmark.Package)
+}
+
+type WorkerPool struct {
+	Size   int
+	Worker PackageWorker
+}
+
+func (pool *WorkerPool) Process(ctx context.Context, packages []benchmark.Package) []schema.Verdict {
+	jobs := make(chan benchmark.Package, *maxConcurrency)
+	results := make(chan schema.Verdict)
+	bar := pb.StartNew(len(packages))
```

Comment:
I think the progress bar logic should be moved out of this abstraction and a callback func() be added to the struct to be called on each iteration. the bar would then be managed by the caller and updated via that callback

Reply:
Done.